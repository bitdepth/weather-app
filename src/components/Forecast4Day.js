import React from 'react';

import {
    Flex,
    Text,
} from '@rebass/emotion';


const DayForecast = ({day, description, low, high}) => (
    <Flex
        p={3}
        width={1 / 2}
        color='magenta'
        bg='white'
        justifyContent='space-between'
    >

        <Text
            color='black'
            textAlign='center'
        >
            {day}
        </Text>
        <Text
            textAlign='center'
            color='black'
        >
            {description}
        </Text>
        <Text
            textAlign='center'
            color='black'
        >
            {high} &deg;
        </Text>
        <Text
            textAlign='center'
            color='black'
        >
            {low} &deg;
        </Text>

    </Flex>
);


export function Forecast4Day({forecast = []}) {
    return (
        <>
            {forecast.map((item) => <DayForecast key={item.day} {...item} />)}
        </>
    );
}
