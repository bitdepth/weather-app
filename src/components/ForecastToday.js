import React from 'react';
import T from 'prop-types';

import {
    Box,
    Text,
} from '@rebass/emotion';


export function ForecastToday({city, description, high}) {

    return (
        <Box
            p={3}
            color='magenta'
            width={1 / 2}
            bg='white'
            data-testid='weather-today'
        >

            <Text
                fontSize={[4, 5, 6]}
                color='black'
                fontWeight='bold'
                textAlign='center'
            >
                {city}
            </Text>
            <Text
                textAlign='center'
                color='black'
            >
                {description}
            </Text>
            <Text
                fontSize={[4, 5, 6]}
                textAlign='center'
                color='black'
            >
                {high} &deg;
            </Text>

        </Box>
    );
}

ForecastToday.propTypes = {
    city: T.string.isRequired,
    high: T.number.isRequired,
    description: T.string.isRequired,
    day: T.string,
    low: T.number,
};
