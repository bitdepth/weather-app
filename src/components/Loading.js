import {
    Box,
    Text,
} from '@rebass/emotion';
import React from 'react';

export function Loading() {

    return (
        <Box
            p={3}
            width={1 / 2}
            bg='white'
            data-testid='loading'
        >
            <Text
                textAlign='center'
                color='black'
            >
                Loading weather data...
            </Text>

        </Box>
    );
}