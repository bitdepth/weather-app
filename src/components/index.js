export { ForecastToday } from './ForecastToday';
export { Forecast4Day } from './Forecast4Day';