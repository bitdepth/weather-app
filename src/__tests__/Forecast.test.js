import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';
import { ForecastToday } from '../components/';

// jest.mock('axios');

it('sums numbers', () => {
    it('should fetch a list of tasks', () => {
        // const getSpy = jest.spyOn(axios, 'get');
        const forecastInstance = shallow(
            <ForecastToday />
        );
        expect(1+1).toEqual(2);
    });
});