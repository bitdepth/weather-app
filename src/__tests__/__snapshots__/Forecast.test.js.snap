// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Forecast4Day matches snapshot 1`] = `
<Forecast4Day
  forecast={
    Array [
      Object {
        "day": "Tue",
        "description": "Light Rain",
        "high": 14,
        "low": 5,
      },
      Object {
        "day": "Wed",
        "description": "Scatterred Clouds",
        "high": 17,
        "low": 6,
      },
      Object {
        "day": "Thurs",
        "description": "Moderate Rain",
        "high": 15,
        "low": 3,
      },
      Object {
        "day": "Fri",
        "description": "Broken Clouds",
        "high": 17,
        "low": 5,
      },
    ]
  }
>
  <Styled(div)
    alignItems="center"
    data-testid="weather-4day"
    flexDirection="column"
    width={1}
  >
    <div
      className="css-1jqcmjx"
      data-testid="weather-4day"
      width={1}
    >
      <DayForecast
        day="Tue"
        description="Light Rain"
        high={14}
        key="Tue"
        low={5}
      >
        <Styled(div)
          bg="white"
          color="magenta"
          justifyContent="space-between"
          p={3}
          width={0.5}
        >
          <div
            className="css-14dzn46"
            color="magenta"
            width={0.5}
          >
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Tue
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Light Rain
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                14
                 °
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                5
                 °
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </DayForecast>
      <DayForecast
        day="Wed"
        description="Scatterred Clouds"
        high={17}
        key="Wed"
        low={6}
      >
        <Styled(div)
          bg="white"
          color="magenta"
          justifyContent="space-between"
          p={3}
          width={0.5}
        >
          <div
            className="css-14dzn46"
            color="magenta"
            width={0.5}
          >
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Wed
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Scatterred Clouds
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                17
                 °
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                6
                 °
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </DayForecast>
      <DayForecast
        day="Thurs"
        description="Moderate Rain"
        high={15}
        key="Thurs"
        low={3}
      >
        <Styled(div)
          bg="white"
          color="magenta"
          justifyContent="space-between"
          p={3}
          width={0.5}
        >
          <div
            className="css-14dzn46"
            color="magenta"
            width={0.5}
          >
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Thurs
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Moderate Rain
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                15
                 °
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                3
                 °
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </DayForecast>
      <DayForecast
        day="Fri"
        description="Broken Clouds"
        high={17}
        key="Fri"
        low={5}
      >
        <Styled(div)
          bg="white"
          color="magenta"
          justifyContent="space-between"
          p={3}
          width={0.5}
        >
          <div
            className="css-14dzn46"
            color="magenta"
            width={0.5}
          >
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Fri
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                Broken Clouds
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                17
                 °
              </div>
            </Styled(div)>
            <Styled(div)
              color="black"
              textAlign="center"
            >
              <div
                className="css-1xq5fjj"
                color="black"
              >
                5
                 °
              </div>
            </Styled(div)>
          </div>
        </Styled(div)>
      </DayForecast>
    </div>
  </Styled(div)>
</Forecast4Day>
`;

exports[`Component: ForecastToday matches snapshot 1`] = `
<ForecastToday
  city="Glasgow"
  day="Mon"
  description="Light Rain"
  high={16}
  low={4}
>
  <Styled(div)
    bg="white"
    color="magenta"
    data-testid="weather-today"
    p={3}
    width={0.5}
  >
    <div
      className="css-z4r8di"
      color="magenta"
      data-testid="weather-today"
      width={0.5}
    >
      <Styled(div)
        color="black"
        fontSize={
          Array [
            4,
            5,
            6,
          ]
        }
        fontWeight="bold"
        textAlign="center"
      >
        <div
          className="css-1qln6of"
          color="black"
          fontSize={
            Array [
              4,
              5,
              6,
            ]
          }
          fontWeight="bold"
        >
          Glasgow
        </div>
      </Styled(div)>
      <Styled(div)
        color="black"
        textAlign="center"
      >
        <div
          className="css-1xq5fjj"
          color="black"
        >
          Light Rain
        </div>
      </Styled(div)>
      <Styled(div)
        color="black"
        fontSize={
          Array [
            4,
            5,
            6,
          ]
        }
        textAlign="center"
      >
        <div
          className="css-o105s2"
          color="black"
          fontSize={
            Array [
              4,
              5,
              6,
            ]
          }
        >
          16
           °
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</ForecastToday>
`;
